include/drivers/gpio.h

```diff
@@ -602,7 +652,21 @@ static inline int gpio_pin_configure(struct device *port, u32_t pin,
 		 || (flags & GPIO_OUTPUT) != 0,
 		 "Output needs to be enabled to be initialized low or high");
 
-	return gpio_config(port, GPIO_ACCESS_BY_PIN, pin, flags);
+	ret = gpio_config(port, GPIO_ACCESS_BY_PIN, pin, flags);
+
+	if (ret == 0) {
+		if ((flags & GPIO_ACTIVE_LOW) != 0) {
+			data->invert |= BIT(pin);
+		} else {
+			data->invert &= ~BIT(pin);
+		}
+		if (api->pin_interrupt_configure) {
+			ret = z_impl_gpio_pin_interrupt_configure(port, pin,
+								  flags);
+		}
+	}
```

Comment:
To avoid nesting this part could be written as
```
	ret = gpio_config(port, GPIO_ACCESS_BY_PIN, pin, flags);
	if (ret != 0) {
		return ret;
	}

	if ((flags & GPIO_ACTIVE_LOW) != 0) {
		data->invert |= BIT(pin);
	} else {
		data->invert &= ~BIT(pin);
	}
	if (api->pin_interrupt_configure) {
		ret = z_impl_gpio_pin_interrupt_configure(port, pin, flags);
	}
```